ddtrace/contrib/internal/openai/_endpoint_hooks.py

```diff
@@ -715,3 +719,52 @@ def _record_response(self, pin, integration, span, args, kwargs, resp, error):
         else:
             span.set_metric("openai.response.total_bytes", getattr(resp, "total_bytes", 0))
         return resp
+
+
+class _ResponseHook(_BaseCompletionHook):
+    _request_arg_params = ("api_key", "api_base", "api_type", "request_id", "api_version", "organization")
+    _request_kwarg_params = (
+        "model",
+        "include",
+        "instructions",
+        "max_output_tokens",
+        "metadata",
+        "parallel_tool_calls",
+        "previous_response_id",
+        "reasoning",
+        "service_tier",
+        "store",
+        "stream",
+        "temperature",
+        "text",
+        "tool_choice",
+        "tools",
+        "top_p",
+        "truncation",
+        "user",
+    )
+    _response_attrs = ("created_at", "id", "model")
+    ENDPOINT_NAME = "responses"
+    HTTP_METHOD_TYPE = "POST"
+    OPERATION_ID = "createResponse"
+
+    def _record_response(self, pin, integration, span, args, kwargs, resp, error):
+        resp = super()._record_response(pin, integration, span, args, kwargs, resp, error)
+
+        if not resp:
+            return resp
+
+        if kwargs.get("stream") and error is None:
+            if _is_async_generator(resp):
+                return TracedOpenAIAsyncResponseStream(resp, integration, span, kwargs, is_completion=False)
+            elif _is_generator(resp):
+                return TracedOpenAIResponseStream(resp, integration, span, kwargs, is_completion=False)
+            return resp
+
+        if not kwargs.get("stream") and error is None:
```

Comment:
are there any cases that wouldn't satisfy this condition at this point? what's the purpose of this check?